active/CornerRadius/CornerRadiusFilterConverter.md

```diff
@@ -0,0 +1,37 @@
+# Background
+
+Some controls need to turn on/off part of their rounded corners depending on their states.
+
+e.g. When CommandBarFlyout opens up its secondary menu, primary and secondary menus are clipped together we need to clear rounded corners on bottom of primary menu and top of secondary menu to close the gaps.
+
+![CommandBarFlyout CornerRadius example](images/commandbarflyout-cornerradius-example.png)
+
+You can't set individual values of a CornerRadius object in markup according to our [doc](https://docs.microsoft.com/en-us/uwp/api/Windows.UI.Xaml.CornerRadius#notes-on-xaml-syntax). We need a way to get non-uniform corner radius values to implement the design above.
+# Description
+
+CornerRadiusFilterConverter helps us filter CornerRadius values on specific directions and set values to 0 for the rest.
+
+# API Details
+
+```
+namespace Microsoft.UI.Xaml.Controls.Primitives
+{
+    [bindable]
+    [webhosthidden]
+    [default_interface]
+    runtimeclass CornerRadiusFilterConverter : Windows.UI.Xaml.Data.IValueConverter
+    {
+        CornerRadiusFilterConverter();
+    };
+}
+```
+
+# Examples
+
+```XAML
+  <Border CornerRadius="{Binding Source={ThemeResource ControlCornerRadius}, Converter={StaticResource CornerRadiusFilterConverter}, ConverterParameter=Top}" />

```

Comment:
Why not just create two new resources: ControlCornerRadiusTop/Bottom?

Reply:
We are trying to automatically extract correct non-uniform values from the value that user passes in, so we can't just hard code it.

If they set CornerRadius="x" on CommandBarFlyout, when secondary menu is not shown, primary menu CornerRadius is x,x,x,x, but when secondary menu opens up, primary menu CornerRadius becomes x,x,0,0.